code/src/sixsq/nuvla/server/resources/spec/voucher.cljc

```diff
@@ -0,0 +1,189 @@
+(ns sixsq.nuvla.server.resources.spec.voucher
+  (:require
+    [clojure.spec.alpha :as s]
+    [sixsq.nuvla.server.resources.spec.common :as cimi-common]
+    [sixsq.nuvla.server.resources.spec.core :as cimi-core]
+    [sixsq.nuvla.server.util.spec :as su]
+    [spec-tools.core :as st]))
+
+
+(s/def ::owner ::cimi-common/resource-link)
+(s/def ::user ::cimi-common/resource-link)
+
+(s/def ::amount
+  (-> (st/spec double)
+      (assoc :name "amount"
+             :json-schema/type "double"
+             :json-schema/editable true
+
+             :json-schema/display-name "voucher amount"
+             :json-schema/description "amount of the voucher"
+             :json-schema/group "body"
+             :json-schema/order 20
+             :json-schema/hidden false
+             :json-schema/sensitive false)))
+
+
+(s/def ::currency
+  (-> (st/spec ::cimi-core/nonblank-string)
+      (assoc :name "currency"
+             :json-schema/type "string"
+             :json-schema/editable true
+
+             :json-schema/display-name "currency"
+             :json-schema/description "currency for the voucher amount"
+             :json-schema/group "body"
+             :json-schema/order 21
+             :json-schema/hidden false
+             :json-schema/sensitive false)))
+
+
+(s/def ::expiration-date
```

Comment:
`expiry` would be a better name. An `expiry` attribute with similar semantics is used elsewhere.  In any case, `expiration-date` is misleading as the field contains a timestamp.

Reply:
done